src/services/creatorStaking/creatorsList.ts

```diff
@@ -0,0 +1,46 @@
+import { CreatorStakingProps } from "."
+import Cache from "../../cache"
+import { ONE_HOUR } from "../../constant"
+import { parseStringValue } from "../utils"
+
+type RegisteredStatus = 'Active' | 'Inactive'
+
+type RegisteredCreator = {
+  spaceId: string
+  stakeholder: string
+  status: RegisteredStatus
+}
+
+const creatorsListCache = new Cache('creators-list', ONE_HOUR)
+
+export const getCreatorsList = async ({
+  apis
+}: CreatorStakingProps): Promise<RegisteredCreator[]> => {
+  const api = apis.subsocial
+
+  if (!api) return []
+
+  const needUpdate = creatorsListCache.needUpdate
+
+  const forceUpdate = needUpdate && (await needUpdate())
+  const cacheData = await creatorsListCache.get()
+
+  if (!cacheData || forceUpdate) {
+    const creatorsCodec = await api.query.creatorStaking.registeredCreators.entries()
+
+    const creators = creatorsCodec.map(([key, value]) => {
+      const spaceId = key.toHuman()[0] as string
```

Comment:
from what I know, isn't there toPrimitive() which I think is better for this? so you don't need to remove the , manually

Reply:
toPrimitive is not working correctly